### PR TITLE
TRM-22121 Write a rest endpoint to search disease

### PR DESCRIPTION
**Pull request by sahmad-ebi**
Please review the changes.

### CODE REVIEW

File: support-data-rest/src/main/java/uk/ac/ebi/uniprot/api/disease/validator/DiseaseFields.java [hunk 1]
```diff
@@ -0,0 +1,55 @@
...
+    }
+
+    DiseaseFields(String solrFieldName, boolean indexed, boolean stored, boolean visible) {
+        this.solrFieldName = solrFieldName;
+        this.indexed = indexed;
+        this.stored = stored;
+        this.visible = visible;
+    }
+
+    public String getSolrFieldName(){
```
Comment (eddturner) [inline support-data-rest/src/main/java/uk/ac/ebi/uniprot/api/disease/validator/DiseaseFields.java]: if we're using lombok can we use a @Getter annotation here? Not sure if this works for enums.

File: support-data-rest/src/test/java/uk/ac/ebi/uniprot/repository/DataStoreTestConfig.java [hunk 2]
```diff
@@ -0,0 +1,40 @@
...
+import java.io.IOException;
+
+@TestConfiguration
+public class DataStoreTestConfig {
+
+    @Bean(destroyMethod = "close")
+    public DataStoreManager dataStoreManager() throws IOException {
+        SolrDataStoreManager sdsm = new SolrDataStoreManager();
+        DataStoreManager dm = new DataStoreManager(sdsm);
+        //ClosableEmbeddedSolrClient solrClient = new ClosableEmbeddedSolrClient(SolrCollection.disease);
```
Comment (eddturner) [inline support-data-rest/src/test/java/uk/ac/ebi/uniprot/repository/DataStoreTestConfig.java]: can delete comments

File: support-data-rest/src/main/java/uk/ac/ebi/uniprot/api/disease/DiseaseDocumentToDiseaseConverter.java [hunk 3]
```diff
@@ -0,0 +1,28 @@
...
+import uk.ac.ebi.uniprot.cv.disease.Disease;
+import uk.ac.ebi.uniprot.json.parser.disease.DiseaseJsonConfig;
+import uk.ac.ebi.uniprot.search.document.disease.DiseaseDocument;
+
+import java.io.IOException;
+import java.util.function.Function;
+
+@Slf4j
+@Service
+public class DiseaseDocumentToDiseaseConverter implements Function<DiseaseDocument, Disease> {
```
Comment (eddturner) [inline support-data-rest/src/main/java/uk/ac/ebi/uniprot/api/disease/DiseaseDocumentToDiseaseConverter.java]: corresponding test for this class?

Comment (sahmad-ebi) [reply support-data-rest/src/main/java/uk/ac/ebi/uniprot/api/disease/DiseaseDocumentToDiseaseConverter.java]: Edd,
I have added many tests in DiseaseTest to test the ObjectMapper. That was the reason i had not written the test cases for this class. But i have added a test class with one test case the DiseaseDocumetnToDiseaseConverter class.

**Pull request merged**